modules/external-api/src/main/java/org/opencastproject/external/util/SchedulingUtils.java

```diff
@@ -256,10 +256,20 @@ public static SchedulingInfo of(JSONObject json) {
       final SchedulingInfo schedulingInfo = new SchedulingInfo();
       final String startDate = (String) json.get(JSON_KEY_START_DATE);
       final String endDate = (String) json.get(JSON_KEY_END_DATE);
-      final String durationString = (String) json.get(JSON_KEY_DURATION);
       final String agentId = (String) json.get(JSON_KEY_AGENT_ID);
       final JSONArray inputs = (JSONArray) json.get(JSON_KEY_INPUTS);
       final String rrule = (String) json.get(JSON_KEY_RRULE);
+
+      // Special handling because the original implementation required String but now we require long
+      final Object durationObj = json.get(JSON_KEY_DURATION);
+      String durationString = null;
```

Comment:
Instead of checking for the object type and handling things differently based on the typüe, couldn't you just use

```java
final String durationString = String.valueOf(durationObj);
```
That should always work, regardless of the type being a String, a Long, a Float or whatever else.

Reply:
EDIT: To handle `null` properly, this whould be better:

```java
… = Objects.toString(json.get(JSON_KEY_DURATION), null);
```

Reply:
That is indeed more elegant, thank you.

Reply:
also fixed a cosmetic double quote in the message.